### PR TITLE
ARC-1743 - fetch reviewers on pull backfill

### PR DESCRIPTION
**Pull request by joshkay10**
**What's in this PR?**
Get the reviewers when we do a backfill for pulls!
Combine the sync and webhook transform function to one to rule them all.

**Why**
We have customers losing reviewer data when they run a backfill, not ideal.

Reviewers from webhook events:
Screen Shot 2022-11-17 at 9 45 39 AM

Reviewers after a backfill
Screen Shot 2022-11-17 at 9 47 19 AM

### CODE REVIEW

File: src/transforms/transform-pull-request.ts [hunk 1]
```diff
@@ -52,66 +52,33 @@ export const transformPullRequest = async (gitHubInstallationClient: GitHubInsta
 	const prBody = await booleanFlag(BooleanFlags.ASSOCIATE_PR_TO_ISSUES_IN_BODY, true) ? body : "";
 	const issueKeys = jiraIssueKeyParser(`${prTitle}\n${head.ref}\n${prBody}}`);
 
-	const logPayload = {
-		prTitle: prTitle || "none",
-		repoName: head?.repo?.name || "none",
```
Comment (joshkay10) [inline src/transforms/transform-pull-request.ts:57]: Removing UGC/PII and just tidying up a bit

File: src/transforms/transform-pull-request.ts [hunk 2]
```diff
@@ -52,66 +52,33 @@ export const transformPullRequest = async (gitHubInstallationClient: GitHubInsta
...
-							url: `${pullRequest.head.repo.html_url}/commit/${pullRequest.head.sha}`
-						},
-						id: getJiraId(pullRequest.head.ref),
-						issueKeys,
-						name: pullRequest.head.ref,
-						url: `${pullRequest.head.repo.html_url}/tree/${pullRequest.head.ref}`,
-						updateSequenceId: Date.now()
-					}
-				],
+		branches: await getBranches(gitHubInstallationClient, pullRequest, issueKeys),
```
Comment (joshkay10) [inline src/transforms/transform-pull-request.ts:66]: Just moved this block to a function for readability

File: src/transforms/transform-pull-request.ts [hunk 3]
```diff
@@ -52,66 +52,33 @@ export const transformPullRequest = async (gitHubInstallationClient: GitHubInsta
...
 		pullRequests: [
 			{
 				// Need to get full name from a REST call as `pullRequest.user.login` doesn't have it
 				author: getJiraAuthor(pullRequest.user, await getGithubUser(gitHubInstallationClient, pullRequest.user?.login)),
-				commentCount: pullRequest.comments,
-				destinationBranch: pullRequest.base.ref,
+				commentCount: pullRequest.comments || 0,
```
Comment (joshkay10) [inline src/transforms/transform-pull-request.ts:71]: The default values come from the sync transform variation.

File: src/transforms/transform-pull-request.ts [hunk 4]
```diff
@@ -52,66 +52,33 @@ export const transformPullRequest = async (gitHubInstallationClient: GitHubInsta
...
-	const logPayload = {
-		prTitle: prTitle || "none",
-		repoName: head?.repo?.name || "none",
-		prRef: pullRequest.head.ref || "none"
-	};
-
 	if (isEmpty(issueKeys) || !head?.repo) {
-		log?.info(logPayload, "Ignoring pullrequest since it has no issue keys or repo");
+		log?.info({
+			pullRequestReference: pullRequest.head.ref,
```
Comment (bgvozdev) [inline src/transforms/transform-pull-request.ts:57]: Is it a branch name?

File: src/sync/pull-request.ts [hunk 5]
```diff
@@ -87,14 +87,8 @@ export const getPullRequestTask = async (
...
-				const ghUser = await getGithubUser(gitHubInstallationClient, prDetails?.user.login
-				);
-				const data = await transformPullRequest(
-					{ pullRequest: pull, repository },
-					prDetails,
-					gitHubInstallationClient.baseUrl,
-					ghUser
-				);
+				const	reviews = await getPullRequestReviews(gitHubInstallationClient, repository, pull, logger);
+				const data = await transformPullRequest(gitHubInstallationClient, prDetails, reviews, logger);
```
Comment (bgvozdev) [inline src/sync/pull-request.ts:91]: Should we maybe feature-flag this change?

File: src/github/pull-request.ts [hunk 6]
```diff
@@ -36,7 +37,7 @@ export const pullRequestWebhookHandler = async (context: WebhookContext, jiraCli
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 	let reviews: Octokit.PullsListReviewsResponse = [];
 	try {
-		reviews = await getReviews(gitHubInstallationClient, owner, repoName, pull_request.number);
+		reviews = await getPullRequestReviews(gitHubInstallationClient, context.payload.repository, pull_request, context.log);
```
Comment (bgvozdev) [inline src/github/pull-request.ts:40]: Same here: should we feature-flag this change?

File: src/sync/pull-request.ts (hunk 5 above)
Comment (rachellerathbone) [reply src/sync/pull-request.ts:91]: Think I agree with Boris. Better to mitigate the chances of it going pear-shaped.

Comment (joshkay10) [reply src/sync/pull-request.ts:91]: Yeah thats not unreasonable, will slap in a FF, and have the old transform as false and this new one on true.
please hold.

File: src/github/pull-request.ts (hunk 6 above)
Comment (joshkay10) [reply src/github/pull-request.ts:41]: Flag added, updated tests to test true/false paths

File: src/transforms/transform-pull-request.ts (hunk 4 above)
Comment (joshkay10) [reply src/transforms/transform-pull-request.ts]: tiz, renamed for clarity ⚡

File: src/sync/pull-request.ts (hunk 5 above)
Comment (joshkay10) [reply src/sync/pull-request.ts]: Flag added, updated tests to test true/false paths

File: src/transforms/transform-pull-request.ts [hunk 7]
```diff
@@ -52,66 +52,33 @@ export const transformPullRequest = async (gitHubInstallationClient: GitHubInsta
...
-	const logPayload = {
-		prTitle: prTitle || "none",
-		repoName: head?.repo?.name || "none",
-		prRef: pullRequest.head.ref || "none"
-	};
-
 	if (isEmpty(issueKeys) || !head?.repo) {
-		log?.info(logPayload, "Ignoring pullrequest since it has no issue keys or repo");
+		log?.info({
+			pullRequestBranchName: pullRequest.head.ref,
```
Comment (bgvozdev) [inline src/transforms/transform-pull-request.ts:57]: Will it be hashed?

Comment (joshkay10) [reply src/transforms/transform-pull-request.ts:57]: ha, did not even register that it was UGC. no no it will not
I shall drop it

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**